pandas/core/series.py

```diff
@@ -2281,36 +2285,62 @@ def _binop(self, other, func, level=None, fill_value=None):
 
     def combine(self, other, func, fill_value=None):
         """
-        Perform elementwise binary operation on two Series using given function
-        with optional fill value when an index is missing from one Series or
-        the other
+        Combine the Series with a Series or scalar according to `func`.
+
+        Combine the Series and `other` using `func` to perform elementwise
+        selection for combined Series.
+        `fill_value` is assumed when value is missing at some index
+        from one of the two objects being combined.
 
         Parameters
         ----------
-        other : Series or scalar value
+        other : Series or scalar
+            The value(s) to be combined with the `Series`.
         func : function
-            Function that takes two scalars as inputs and return a scalar
-        fill_value : scalar value
-            The default specifies to use the appropriate NaN value for
-            the underlying dtype of the Series
+            Function that takes two scalars as inputs and returns an element.
+        fill_value : scalar, optional
+            The value to assume when an index is missing from
+            one Series or the other. The default specifies to use the
+            appropriate NaN value for the underlying dtype of the Series.
 
         Returns
         -------
-        result : Series
+        Series
+            The result of combining the Series with the other object.
+
+        See Also
+        --------
+        Series.combine_first : Combine Series values, choosing the calling
+                               Series' values first.
 
         Examples
         --------
         >>> s1 = pd.Series([1, 2])
-        >>> s2 = pd.Series([0, 3])
-        >>> s1.combine(s2, lambda x1, x2: x1 if x1 < x2 else x2)
-        0    0
-        1    2
+        >>> s2 = pd.Series([0, 3, 4])
+        >>> s1.combine(s2, lambda x1, x2: x1 if x1 > x2 else x2)
+        0    1
+        1    3
+        2    4
+        dtype: int64
+
+        >>> arms = pd.Series({'dog':2,'cat': 2,'mouse': 2})
+        >>> legs = pd.Series({'dog':2,'cat': 2})
+        >>> arms
+        dog      2
+        cat      2
+        mouse    2
```

Comment:
I'd say all those animals have 4 legs and 0 arms. Use correct data, and different values (spider, monkey...)

Reply:
Does this example look fine?
```
>>> arms = pd.Series({'starfish': 5, 'kangaroo': 2})
>>> legs = pd.Series({'dog': 4,'kangaroo': 2})
>>> arms.combine(legs, lambda x, y: x+y, fill_value=0)
dog         4
kangaroo    4
starfish    5
dtype: int64
>>> arms.combine(legs, lambda x, y: x+y, fill_value=10)
dog         14
kangaroo     4
starfish    15
dtype: int64
>>>
```

Reply:
I could not find any alternative to defining a `lambda`, since the next alternative was to use `+` itself, which cannot be done as a functor afaik

Reply:
Besides the example not following pep8 again, you may not be able to use a sum without a `lambda`, but as I said, you can use the builtin `max` function.

In the first example, remove completely the `fill_value` param, and in the second use a example that makes sense.

Also, add a short description before each case, explaining what is the goal of what you're showing. pandas does not contain functions or methods that are not useful for real-world examples. And in the documentation examples we should show those.

Reply:
The purpose of combine() is to merge two Series into one.
So, I need 2 Series with some conflicting values[that use `func` to resolve conflict], and some NOT CONFLICTING ones[that could make use of `fill_value` to assume a value from the lacking Series]
I will use `max` for an example I just thought!

Reply:
Consider 2 Datasets `data_A` and `data_B` containing highest clocked speeds of different birds.
```
>>> data_A = pd.Series({'Falcon': 330, 'Eagle': 160, 'Swift': 105, 'Duck': 10})
>>> data_B = pd.Series({'Falcon': 345, 'Eagle': 200, 'Swift': 100})
```
Now, to combine the two datasets and view the highest speeds of the birds across the
two datasets
```
>>> data_A.combine(data_B, max)
Duck       10
Eagle     200
Falcon    345
Swift     105
dtype: int64
```
Considering an average value of birds' speed to be ~25 mph,
```
>>> data_A.combine(data_B, max, fill_value=25)
Duck       25
Eagle     200
Falcon    345
Swift     105
dtype: int64
>>>
```

Reply:
will this one work, @datapythonista ? The "average speed" was assumed after reading [from here](https://web.stanford.edu/group/stanfordbirds/text/essays/How_Fast.html)

Reply:
That looks much better, yes. I'd name the Series `s1` and `s2` following our standards, and also have the animal names in all lower case.

And I don't quite understand the `fill_value` part. I guess I'm misunderstanding something, but if:
```
>>> max(float('NaN'), 25)
nan
```
I would expect that in the first example `duck` has a value of `nan`. And in that case I'd use `fill_value=0` to get the known value. I don't quite lack making up an average speed. But I guess the function doesn't work as I think.

Can you do a bit of research and move the examples to the PR, so I can add new comments in a review.

Thanks!

Reply:
@datapythonista look something funny!
```
>>> max(30,float('nan'))
30
>>> max(float('nan'),30)
nan
```

Reply:
@datapythonista `fill_value` is used as the default value for the missing key in the corresponding series _before_ the comparison is made.

So in the example above, since `data_B` doesn't have `Duck`, the comparison is made as if `data_B['Duck'] = fill_value = 40`

Reply:
> I would expect that in the first example `duck` has a value of `nan`.

[This answer on StackOverflow](https://stackoverflow.com/questions/47788361/why-does-max-sometimes-return-nan-and-sometimes-ignores-it) clears this doubt for me!

Reply:
@datapythonista please read this !